lib/vf_common.cpp

```diff
@@ -93,6 +93,20 @@ namespace ValueFlow
         return true;
     }
 
+    long long truncateIntValue(long long value, size_t value_size, const ValueType::Sign dst_sign)
+    {
+        if (value_size == 0)
+            return value;
+
+        const MathLib::biguint unsignedMaxValue = std::numeric_limits<MathLib::biguint>::max() >> (value_size < sizeof(unsignedMaxValue) ? ((sizeof(unsignedMaxValue) - value_size) * 8) : 0);
```

Comment:
I don't understand this calculation.

The unsigned max value when value_size is 1 should be 255, shouldn't it?

Reply:
If value_size is 1, `std::numeric_limits<MathLib::biguint>::max()` is shifted by 56 because unsignedMaxValue is 64 bits. So you end up with 255.
However, this line was a bit complex because it assumed that value_size could be set to 0. Since we now avoid truncating if value_size is equal to 0, I simplified this line.